warp/Network/Wai/Handler/Warp/HTTP2/Receiver.hs

```diff
@@ -232,9 +232,12 @@ control FrameGoAway _ _ Context{controlQ} = do
 
 control FrameWindowUpdate header bs Context{connectionWindow} = do
     WindowUpdateFrame n <- guardIt $ decodeWindowUpdateFrame header bs
-    !w <- (n +) <$> atomically (readTVar connectionWindow)
+    !w <- atomically $ do
```

Comment:
I worry about space leak. If you don't have a strong reason to use lazy evaluation, I would recommend the  bang pattern.